tests/run-make/autodiff/type-trees/type-analysis/array/rmake.rs

```diff
@@ -0,0 +1,27 @@
+//@ needs-enzyme
+
+use std::fs;
+
+use run_make_support::{llvm_filecheck, rfs, rustc};
+
+fn main() {
+    // Compile the Rust file with the required flags, capturing both stdout and stderr
+    let output = rustc()
+        .input("array.rs")
+        .arg("-Zautodiff=Enable,PrintTAFn=callee")
+        .arg("-Zautodiff=NoPostopt")
+        .arg("-Copt-level=3")
+        .arg("-Clto=fat")
+        .arg("-g")
+        .run();
+
+    let stdout = output.stdout_utf8();
+    let stderr = output.stderr_utf8();
+
+    // Write the outputs to files
+    fs::write("array.stdout", stdout).unwrap();
+    fs::write("array.stderr", stderr).unwrap();
```

Comment:
Suggestion: `run_make_support` has an fs helper module re-export as `run_make_support::rfs` (https://doc.rust-lang.org/nightly/nightly-rustc/run_make_support/rfs/index.html), which will produce a panic incl. the path, so this can be `rfs::write("array.stdout", stdout);` (https://doc.rust-lang.org/nightly/nightly-rustc/run_make_support/rfs/fn.write.html).

Ditto on other test recipes.